api_tests/src/ledgers/geth_instance.ts

```diff
@@ -0,0 +1,224 @@
+import { ChildProcess, spawn } from "child_process";
+import waitForLogMessage from "../wait_for_log_message";
+import { existsAsync, openAsync } from "../utils";
+import { promises as asyncFs } from "fs";
+import getPort from "get-port";
+import { Logger } from "log4js";
+import { LedgerInstance } from "./index";
+import findCacheDir from "find-cache-dir";
+import download from "download";
+import { platform } from "os";
+import chmod from "chmod";
+import * as path from "path";
+
+export class GethInstance implements LedgerInstance {
+    private process: ChildProcess;
+    // The parameter --networkid does not seem to have an effect for dev chain.
+    // Nevertheless we define it so that it is obvious where the chain_id is coming from
+    public readonly CHAIN_ID = 1337;
+
+    public static async new(dataDir: string, pidFile: string, logger: Logger) {
+        return new GethInstance(
+            dataDir,
+            pidFile,
+            logger,
+            await getPort({ port: 8545 }),
+            await getPort()
+        );
+    }
+
+    constructor(
+        private readonly dataDir: string,
+        private readonly pidFile: string,
+        private readonly logger: Logger,
+        public readonly rpcPort: number,
+        public readonly p2pPort: number
+    ) {}
+
+    public async start() {
+        const bin = await this.findBinary("1.9.13-cbc4ac26");
+
+        this.logger.info("Using binary", bin);
+
+        await this.createConfigurationFiles();
+
+        this.process = spawn(
+            bin,
+            [
+                `--dev`,
+                `--dev.period=1`, // generates a block every X seconds
+                `--datadir=${this.dataDir}`,
+                `--networkid=${this.CHAIN_ID}`,
```

Comment:
Can you clarify what you mean with this?

Reply:
The chainID seems to be fixed based on the `dev` profile right? The network ID doesn't matter so I think we don't need to set this. Network ID and Chain ID are not the same thing in Ethereum.

Reply:
Well... it's complicated in ethereum, as usual:D I think for our purpose we can assume they are the same. If we would consider different forks it would be important. 

Found this on the geth wiki: 

> What is the difference between chain id, chain identity, and network id?
> 
> * chain id is used by the chain configuration for signing replay-protected (see EIP155) transactions. It is configured in the "Diehard" upgrade fork.
> * chain identity is a local value, and is not a part of the p2p or blockchain protocols, per se. With default values "mainnet" and "morden", it can be used as an argument to specify a custom chain. When doing so, the configuration file JSON key "identity" must match the containing subdirectory (/datadir/customnet/chain.json) to ensure your configurations are in proper order.
> * network id is used in the p2p protocol to identify valid peers on a given network. Mainnet network id is 1, Morden is 2, and your private net can be any positive integer (although best not 1 or 2, to avoid trying unuseful peers).